### PR TITLE
[FEA] Improve metrics api docs

### PR DESCRIPTION
**Pull request by ktangsali**
# Modulus Pull Request

## Description
Improves API docs for metrics API. 

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Comment by dallasfoster** [comment]
@ktangsali Feel free to take a look.

**Comment by ktangsali** [comment]
> @ktangsali Feel free to take a look.

Thanks @dallasfoster ! Looking good to me. @NickGeneva can you review and approve?

File: docs/api/modulus.metrics.rst [hunk 1]
```diff
@@ -1,9 +1,266 @@
...
    :toctree: generated
 
+General
+---------
+
+.. automodule:: modulus.metrics.general.mse
+    :members:
+    :show-inheritance:
+
+.. _histogram:
```
Comment (NickGeneva) [inline docs/api/modulus.metrics.rst:234]: Is this suppose to be here?

**Review by NickGeneva — APPROVED** [review]

File: docs/api/modulus.metrics.rst [hunk 2]
```diff
@@ -1,9 +1,266 @@
...
 
+.. automodule:: modulus.metrics
+.. currentmodule:: modulus.metrics
+
+Basics
+-------
+
+Modulus provides several general and domain-specific metric calculations you can
+leverage in your custom training and inference workflows. These metrics are optimized to
+oprate on PyTorch tensors.
```
[the remark this answers is not in the document]
Comment (ktangsali) [reply docs/api/modulus.metrics.rst]: Great catch! Fixed :)

File: docs/api/modulus.metrics.rst (hunk 1 above)
Comment (ktangsali) [reply docs/api/modulus.metrics.rst]: Great catch. I was trying something and this was a leftover from the testing. Removed

**Review by akshaysubr — APPROVED** [review]

**Pull request merged**